codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -636,95 +617,101 @@ impl Field {
             }
         };
 
-        // TODO: should we dis-allow indexed fields that are either `id` or derived From fields?
         let is_indexed = indexed_count > 0;
 
         // Parse the field type into UserDefinedFieldType
-        let underlying_scalar = UserDefinedFieldType::from_obj_field_type(
+        let field_type = UserDefinedFieldType::from_obj_field_type(
             &field.field_type,
             &PgTypeModifications::default(),
-        )
-        .get_underlying_scalar();
+        );
 
-        let mut pg_type_modifications = PgTypeModifications::default();
+        // TODO: combine these?
 
-        // Process @precision
-        if let Some(decimal_precision_directive) = decimal_precision_directives.first() {
-            if !matches!(underlying_scalar, GqlScalar::BigInt(_)) {
-                return Err(anyhow!(
-                    "EE215: The precision directive on a field is only suitable for BigInt scalar \
-                     type. Field '{}'",
-                    field.name
-                ));
-            }
-            if decimal_precision_directive.arguments.len() != 1 {
-                return Err(anyhow!(
-                    "EE216: The precision directive on a BigInt should only take a single integer \
-                     argument called 'digits'. Field '{}'",
-                    field.name
-                ));
-            }
-            let (arg_name, arg_value) = decimal_precision_directive.arguments.first().unwrap();
-            if arg_name != "digits" {
-                return Err(anyhow!(
-                    "EE216: The precision directive on a BigInt should only have a 'digits' \
-                     parameter. Unknown parameter '{}'. Field '{}'",
-                    arg_name,
-                    field.name
-                ));
-            }
-            let precision = get_positive_integer(arg_value).context(format!(
-                "Parsing precision.digits directive on BigInt field with field name {}",
-                &field.name
-            ))?;
-            pg_type_modifications.big_int_precision = Some(precision);
-        }
+        let underlying_scalar = field_type.get_underlying_scalar();
```

Comment:
Yeah I think just combine these. It doesn't make sense to have an incomplete version of "field_type" in scope unless it's needed.